src/main/grammars/RustParser.bnf

```diff
@@ -1186,7 +1189,7 @@ UnaryExpr ::= OuterAttr* (box | '-' | '*' | '!' | '&' [raw (mut | const) | mut])
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
-LambdaExpr ::= OuterAttr* [asyncBlock | static] move? LambdaParameters RetType? <<setStmtMode 'StmtMode.OFF'>> Expr <<resetFlags>> {
+LambdaExpr ::= OuterAttr* [asyncBlock | static] move? LambdaParameters RetType? <<setStmtMode 'StmtMode.OFF'>> (<<enterBlockExpr Expr>> | never Expr) <<resetFlags>> {
```

Comment:
@vlad20012 Could you please check this change? I have to reset `CONDITION_MODE` flag, but I'm not sure about other flags.

Reply:
Oh, it look like we should not reset `RESTRICTED_CONST_EXPR_MODE` flag. Otherwise, we'll incorrectly parse `>` as a part of lambda expr in this case:
```rust
struct S<const C: i32>;
type T = S<|| 42 >;
```